pages/[id]/edit.js

```diff
@@ -0,0 +1,177 @@
+import styled from "styled-components";
+import DatePicker from "react-datepicker";
+import "react-datepicker/dist/react-datepicker.css";
+import Map from "../../components/map/index";
+import Modal from "../../components/modal/index";
+import { useState } from "react";
+import Link from "next/link";
+import { useRouter } from "next/router";
+import apiUrl from "next-api-url";
+
+export default function EditCard({
+  startDate,
+  fishName,
+  setStartDate,
+  fishWeight,
+  setFishName,
+  fishLength,
+  setFishWeight,
+  setFishLength,
+  fishLocation,
+  setFishLocation,
+  setLatLng,
+  latlng,
+  fetchedData,
```

Comment:
Why use this global state here? Each page should be responsible for it's own data. You even have a getServerSideProps function here fetching the same data again.